pkg/cmd/server/origin/auth.go

```diff
@@ -355,8 +357,11 @@ func (c *AuthConfig) getAuthenticationHandler(mux cmdutil.Mux, errorHandler hand
 				}
 				passwordSuccessHandler := handlers.AuthenticationSuccessHandlers{c.SessionAuth, redirectSuccessHandler{}}
 
+				loginPath := path.Join(OpenShiftLoginPrefix, identityProvider.Name)
```

Comment:
As a nicety, maybe let the first password auth have the `OpenShiftLoginPrefix`, and require subsequent ones to suffix with their provider name. 99% of deployments will have a single one, and it would be nice to let them have the pretty `/login` url instead of `/login/myidpname`